src/utils.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const { MongoRuntimeError } = require('mongodb');
+
+module.exports = Object.create(null);
+
+module.exports.toLegacy = Symbol.for('@@mdb.callbacks.toLegacy');
+
+/**
+ * @param {any} target
+ * @param {string} symbolName
+ * @returns {symbol}
+ */
+module.exports.getSymbolFrom = (target, symbolName) => {
+  const symbol = Object.getOwnPropertySymbols(target).find(
+    s => s.toString() === `Symbol(${symbolName})`
+  );
+
+  if (symbol == null) {
+    throw new MongoRuntimeError(`Did not find Symbol(${symbolName}) on ${target}`);
+  }
+
+  return symbol;
+};
+
+/**
+ *
+ * @template T
+ * @param {Promise<T>} promise
+ * @param {(error?: Error, result?: T) => void} callback
+ * @param {(res: T) => T} [conversion]
+ * @returns
+ */
+module.exports.maybeCallback = (promise, callback, conversion) => {
```

Comment:
actually while we're here this is basically a mapping function - could we name it accordingly?  just for clarity.  "conversion" isn't immediately clear what it does (I understand where it came from, because it's used to convert between regular and legacy classes).  But there's no reason this is invalid, and then the `conversion` name doesn't really make sense

```typescript
maybeCallback(Promise.resolve(2), null, x => x +1)
```

Reply:
Removed this from the API. I think there's only a few cases where we need this mapping and it doesn't make sense to combine concerns.

I appreciate the use of the identity function, but it has overhead and modifies the promise returned from the driver in every case when just passing it along would work.